### PR TITLE
Various edge case fixes for `toml.stringify`

### PR DESCRIPTION
**Pull request by Aplet123**
Fixes the `stringify` errors in #3402

### CODE REVIEW

File: toml/test.ts [hunk 1]
```diff
@@ -711,3 +711,38 @@ aaaaa = 1
...
+
+Deno.test({
+  name: "[TOML] stringify empty key",
+  fn() {
+    const src = {
+      "": "a",
+      "b": { "": "c" },
+    };
+    const roundTrip = parse(stringify(src));
+    assertEquals(src, roundTrip);
```
Comment (kt3k) [inline toml/test.ts:723]: I think we should directly assert the stringified result, instead of parsing it back to js object

**Review by kt3k — COMMENTED** [review]
Thanks for the fix! Left one comment

**Comment by Aplet123** [comment]
@kt3k I adjusted the tests.

**Review by kt3k — APPROVED** [review]
LGTM. Thanks!
